### ISSUE TITLE
Problems in creating a Python virtual environment

### ISSUE THREAD

**Issue by 2227129479-debug** [VIPL-SLP/XV-SLR_WWW25#5]
My operating system is Ubuntu 22.04.

When I use "conda env create -f environment.yml", there are many prompts related to pip. Tell me that no compatible toolkit was found. There are simply too many such "non-compliant versions" of information.

Could someone please tell me how to download the environment correctly?

**Comment by tttran67** [comment]
Hi, here's an environment that works on Ubuntu 22.04:

absl-py==2.1.0
accelerate==1.1.1
addict==2.4.0
aiofiles==23.2.1
aiohappyeyeballs==2.4.4
aiohttp==3.10.11
aiohttp-retry==2.8.3
aioice==0.10.1
aiortc==1.9.0
aiosignal==1.3.2
aiosqlite==0.21.0
aliyun-python-sdk-core==2.16.0
aliyun-python-sdk-kms==2.16.5
annotated-types==0.7.0
anthropic==0.49.0
anyio==4.9.0
APScheduler==3.11.0
async-timeout==5.0.1
asyncua==1.1.6
attrs==23.2.0
audioread==3.0.1
av==12.3.0
backgroundremover==0.2.8
backoff==2.2.1
bcrypt==4.2.1
blinker==1.7.0
boto3==1.35.60
botocore==1.35.99
cachetools==5.5.2
certifi==2024.2.2
cffi==1.16.0
chardet==3.0.4
charset-normalizer==3.3.2
choreographer==1.0.10
chumpy==0.70
click==8.1.7
colorama==0.4.6
coloredlogs==15.0.1
commandlines==0.4.1
commonmark==0.9.1
conda-pack @ file:///home/conda/feedstock_root/build_artifacts/conda-pack_1731874358732/work
contourpy==1.3.2
crcmod==1.7
cryptography==44.0.0
ctcdecode @ file:///home/pengyuting/ProgramFiles/packages/ctcdecode
cycler==0.12.1
Cython==3.0.12
dataclasses-json==0.6.7
datasets==3.2.0
decorator==4.4.2
defusedxml==0.7.1
dill==0.3.8
distro==1.9.0
dlib==19.24.6
dnspython==2.7.0
docker==7.1.0
einops==0.8.1
evaluate==0.4.3
exceptiongroup==1.2.2
face-recognition==1.3.0
face-recognition-models==0.3.0
fastapi==0.115.12
fastrtc==0.0.24
ffmpeg-python==0.2.0
ffmpy==0.5.0
filelock==3.14.0
filetype==1.2.0
Flask==3.0.2
flatbuffers==24.3.25
fonttools==4.50.0
frozenlist==1.5.0
fsspec==2024.3.1
future==1.0.0
google-crc32c==1.7.1
googletrans==4.0.0rc1
GPUtil==1.4.0
gradio==5.25.2
gradio_client==1.8.0
gradio_webrtc==0.0.31
groovy==0.1.2
h11==0.14.0
h2==3.2.0
hf_transfer==0.1.8
hpack==3.0.0
hsh==1.1.0
hstspreload==2024.6.1
httpcore==1.0.8
httpx==0.28.1
huggingface-hub==0.30.2
humanfriendly==10.0
hyperframe==5.2.0
idna==3.10
ifaddr==0.2.0
imageio==2.36.1
imageio-ffmpeg==0.5.1
importlib_resources==6.5.2
inference==0.49.5
inference-tools==0.14.1
iniconfig==2.1.0
itsdangerous==2.1.2
jax==0.4.26
jaxlib==0.4.26
jieba==0.42.1
Jinja2==3.1.3
jiter==0.10.0
jmespath==0.10.0
joblib==1.3.2
json-tricks==3.17.3
kaleido==1.0.0
kiwisolver==1.4.5
lazy_loader==0.4
libretranslatepy==2.1.1
librosa==0.11.0
llvmlite==0.43.0
logistro==1.1.0
lxml==5.2.2
Markdown==3.8
markdown-it-py==3.0.0
MarkupSafe==2.1.5
marshmallow==3.26.1
matplotlib==3.10.3
mdurl==0.1.2
mediapipe==0.10.11
ml-dtypes==0.4.0
mmcv==2.0.1
mmdet==3.1.0
mmengine==0.10.7
mmpose==1.3.2
model-index==0.1.11
more-itertools==10.5.0
moviepy==1.0.3
mpmath==1.3.0
msgpack==1.1.0
multidict==6.1.0
multiprocess==0.70.16
munkres==1.1.4
mypy_extensions==1.1.0
narwhals==2.1.2
networkx==3.4.2
nltk==3.8.1
numba==0.60.0
numpy==1.23.0
nvidia-ml-py==12.575.51
nvidia-ml-py3==7.352.0
onnxruntime==1.16.3
onnxruntime-gpu==1.16.3
openai==1.79.0
opencv-contrib-python==4.9.0.80
opencv-python==4.9.0.80
opendatalab==0.0.10
openmim==0.3.9
openxlab==0.1.2
opt-einsum==3.3.0
ordered-set==4.1.0
orjson==3.10.16
oss2==2.17.0
packaging==24.0
paho-mqtt==1.6.1
pandas==2.2.3
paramiko==3.5.0
pillow==10.4.0
platformdirs==4.3.8
plotly==6.3.0
pluggy==1.6.0
pooch==1.8.2
portalocker==2.8.2
proglog==0.1.10
prometheus-fastapi-instrumentator==6.0.0
prometheus_client==0.22.0
propcache==0.2.1
protobuf==3.20.3
psutil==5.9.8
py-cpuinfo==9.0.0
pyarrow==18.1.0
pybase64==1.0.2
pycocotools==2.0.8
pycparser==2.22
pycryptodome==3.23.0
pydantic==2.11.3
pydantic-settings==2.7.1
pydantic_core==2.33.1
pydot==2.0.0
pydub==0.25.1
pyee==13.0.0
Pygments==2.19.1
PyJWT==2.10.1
pylibsrtp==0.12.0
pylogix==1.0.5
PyMatting==1.1.13
pymodbus==3.8.3
PyNaCl==1.5.0
pyOpenSSL==25.1.0
pyparsing==3.1.2
PySocks==1.7.1
pytest==8.3.5
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-multipart==0.0.20
pytz==2023.4
PyYAML==6.0.1
redis==5.0.8
regex==2023.12.25
requests==2.32.3
requests-file==2.1.0
requests-toolbelt==1.0.0
rfc3986==1.5.0
rich==13.4.2
ruff==0.11.6
s3transfer==0.10.4
safehttpx==0.1.6
safetensors==0.4.5
scikit-image==0.24.0
scikit-learn==1.5.2
scipy==1.15.3
semantic-version==2.10.0
sentencepiece==0.2.0
shapely==2.0.7
shellingham==1.5.4
simplejson==3.20.1
six==1.16.0
slack_sdk==3.33.5
sniffio==1.3.1
sortedcontainers==2.4.0
sounddevice==0.4.6
soundfile==0.13.1
soxr==0.5.0.post1
spaces==0.36.0
starlette==0.46.2
structlog==24.4.0
supervision==0.25.1
sympy==1.14.0
tabulate==0.9.0
termcolor==3.1.0
terminaltables==3.1.10
threadpoolctl==3.5.0
tifffile==2024.9.20
timm==1.0.12
tldextract==5.1.3
tokenizers==0.20.3
tomli==2.2.1
tomlkit==0.12.0
torch==1.12.1+cu113
torchaudio==0.12.1+cu113
torchtext==0.13.1
torchvision==0.13.1+cu113
tqdm==4.65.2
transformers==4.46.3
translate==3.6.1
truecase==0.0.14
twilio==9.3.8
typer==0.12.5
typing-inspect==0.9.0
typing-inspection==0.4.0
typing_extensions==4.12.2
tzdata==2024.1
tzlocal==5.3.1
urllib3==2.4.0
uvicorn==0.34.2
wait-for2==0.3.2
waitress==3.0.2
warpctc-pytorch==0.1
websockets==12.0
Werkzeug==3.0.1
xtcocotools==1.14.3
xxhash==3.5.0
yapf==0.43.0
yarl==1.18.3
zxing-cpp==2.2.0

you can check for non-compliant packages in the pip list above.
